### ISSUE TITLE
Idemix MSP Folder Structure incompatible with what Fabric expects

### ISSUE THREAD

**Issue by adecaro** [hyperledger/fabric-ca#303]
Currently, Fabric-CA generates an Idemix MSP folder structure and file content that Fabric is not able to load.
Namely:
- The content of `SignerConfig` is encoded in json, Fabric expected proto
- Fabric expects the revocation public key to be stored in a file named `RevocationPublicKey` and `IdemixRevocationPublicKey`.

This git issue is to fix the above issues.

**Comment by arner** [comment]
Why was this change reverted?

**Comment by bestbeforetoday** [comment]
> Why was this change reverted?

It broke the Java client SDK.

**Comment by adecaro** [comment]
Is anyone using the Idemix provided by Java client SDK's ? If no, I would suggest to drop it. No one is going to keep supporting that implementation.

**Comment by bestbeforetoday** [comment]
Yes, people are using it. We already encourage people to use the Fabric Gateway client API instead of the legacy Java SDK, and plan to deprecate the legacy Java SDK once Fabric v2.5 LTS is released. Certainly for Fabric v3 we can break compatibility with the v2.x Idemix implementation.

Producing a new release (v2) of the CA without support for the old style Idemix to go along with Fabric v3 sounds perfectly reasonable. Breaking compatibility with existing client applications or removing capability in a patch release (or even a minor version release) does not seem reasonable.

**Comment by adecaro** [comment]
But Gateway does not support Idemix yet, how are people using idemix moving there?

**Comment by bestbeforetoday** [comment]
The identity credentials and signing implementation are pluggable in the Fabric Gateway client API. Anyone _could_ use the Idemix implementation from the legacy Java SDK with the new client API. I suspect nobody has done that - certainly not that I've seen - and there is currently no Idemix signing implementation provided out-of-the-box with the Fabric Gateway client API so I don't think anyone is using Idemix with that client API.

I really don't understand how that impacts whether it's OK to make breaking changes to the CA in patch releases.

**Comment by bestbeforetoday** [comment]
The fabric-sdk-java client SDK that provided client-side Idemix support was deprecated a while back, and has since been archived. Anyone still using it can stick with an older version of Fabric CA that provides a compatible Idemix implementation. I don't have any objection to these changes going in again. While semver versioning suggests this should trigger a major release version change, we haven't been strict on following semver up to now. At least a minor version change seems like it would flag the incompatibility better to users. If it doesn't break Fabric (or other current Fabric ecosystem projects) then perhaps we could get away with just a patch version change.
